synapse/rest/media/v1/_base.py

```diff
@@ -197,3 +199,64 @@ def __init__(self, server_name, file_id, url_cache=False,
         self.thumbnail_height = thumbnail_height
         self.thumbnail_method = thumbnail_method
         self.thumbnail_type = thumbnail_type
+
+
+def get_filename_from_headers(headers):
+    """
+    Get the filename of the downloaded file by inspecting the
+    Content-Disposition HTTP header.
+
+    Args:
+        headers (twisted.web.http_headers.Headers): The HTTP
+        request headers.
+
+    Returns:
+        A Unicode string of the filename, or None.
+    """
+    content_disposition = headers.get(b"Content-Disposition", [b''])
+
+    # Decode the Content-Disposition header (cgi.parse_header requires
+    # unicode on Python 3, not bytes) the best we can.
+    try:
+        content_disposition = content_disposition[0].decode('utf8')
```

Comment:
surely this will fail if there was no `content_disposition` header?

Reply:
No, because above it does a `.get` with the default argument being a list containing empty bytestring.

Reply:
oh, duh.